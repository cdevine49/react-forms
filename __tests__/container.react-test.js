import React from 'react';
import Container from '../src/components/container';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';

describe("Container", () => {
  test('renders with default props', () => {
    const tree = renderer.create(
      <Container
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  const onChange = jest.fn();

  test('props populate the input tag', () => {
    const tree = renderer.create(
      <Container
        id="test-input"
        className="test-class another class"
        value="Can set a default value"
        type="email"
        onChange={onChange}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("renders with props from containerProps object", () => {
    const tree = renderer.create(
      <Container
        containerProps={{
          id: 'container',
          className: 'classy react',
          style: { margin: "10px" }
        }}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  describe("Label", () => {
    test("label prop acts as Label component's child", () => {
      const tree = renderer.create(
        <Container
          label="Input"
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test("id prop becomes htmlFor", () => {
      const tree = renderer.create(
        <Container
          id="htmlFor"
          label="Input"
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test("labelProps populate label tag", () => {
      const tree = renderer.create(
        <Container
          label="Input"
          labelProps={{
            id: 'label-test',
            className: 'classy react'
          }}
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe("errorMessage prop", () => {
    const message = "Shock horror";

    describe("when error handler return true", () => {
      const container = shallow(
        <Container
          errors={[{ _handle: () => true, message }]}
          />
      );
      const input = container.first().shallow().find('input');

      test("initializes as empty string", () => {
        expect(container.props().errorMessage).toBe('');
      });

      test("equals message on input blur", () => {
        input.simulate('blur');
        expect(container.props().errorMessage).toBe(message);
      });

      test("is empty string on input focused", () => {
        input.simulate('focus');
        expect(container.props().errorMessage).toBe('');
      });
    })

    describe("when error handler returns false", () => {
      const container = shallow(
        <Container
          errors={[{ _handle: () => false, message }]}
          />
      );
      const input = container.first().shallow().find('input');

      test("initializes as empty string", () => {
        expect(container.props().errorMessage).toBe('');
      });

      test("stays empty string on blur", () => {
        input.simulate('blur');
        expect(container.props().errorMessage).toBe('');
      });

      test("stays empty string on input focus", () => {
        input.simulate('focus');
        expect(container.props().errorMessage).toBe('');
      });
    });
  });

  describe("Underline", () => {
    test("renders when passed as prop", () => {
      const tree = renderer.create(
        <Container
          underline="More info about input"
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    test("accepts underlineProps", () => {
      const tree = renderer.create(
        <Container
          underline="More info about input"
          underlineProps={{
            id: "underline",
            className: "new-class test"
          }}
          />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});

//
//
//
//
//
//
//
//
// //
//
