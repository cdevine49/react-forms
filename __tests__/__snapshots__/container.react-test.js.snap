// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Renders with props from containerProps 1`] = `
<div
  className="input-container classy react"
  id="container"
  style={
    Object {
      "margin": "10px",
    }
  }
>
  <input
    className={undefined}
    id={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`Container props populate the input tag 1`] = `
<div
  className="input-container"
>
  <input
    className="test-class another class"
    id="test-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="email"
    value="Can set a default value"
  />
</div>
`;

exports[`Container renders with default props 1`] = `
<div
  className="input-container"
>
  <input
    className={undefined}
    id={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`Container renders with props from containerProps object 1`] = `
<div
  className="input-container classy react"
  id="container"
  style={
    Object {
      "margin": "10px",
    }
  }
>
  <input
    className={undefined}
    id={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`Label Label does not render if label prop is falsey 1`] = `
<div
  className="input-container"
>
  <input
    className={undefined}
    id={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`Label Label prop on container becomes label text 1`] = `
<div
  className="input-container"
>
  <label
    htmlFor={undefined}
  >
    Test Label
  </label>
  <input
    className={undefined}
    id={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`Label Label receives props from container's labelProps prop 1`] = `
<div
  className="input-container"
>
  <label
    className="label-class test"
    htmlFor={undefined}
    id="labelId"
  >
    Test Label
  </label>
  <input
    className={undefined}
    id={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`Props on container populate the input tag 1`] = `
<div
  className="input-container"
>
  <input
    className="test-class another class"
    id="test-input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="email"
    value="Can set a default value"
  />
</div>
`;

exports[`Renders with default props 1`] = `
<div
  className="input-container"
>
  <input
    className={undefined}
    id={undefined}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
    value=""
  />
</div>
`;
